src/api/layers/layers-external.api.js

```diff
@@ -0,0 +1,83 @@
+import axios from 'axios'
+import WmsCapabilities from '@/api/layers/wms-capabilities.class'
+import WmtsCapabilities from '@/api/layers/wmts-capabilities.class'
+import log from '@/utils/logging'
+
+/**
+ * Read and parse WMS GetCapabilities
+ *
+ * @param {string} baseUrl Base URL for the WMS server
+ * @returns {Promise<WmsCapabilities | null>} WMS Capabilities
+ */
+export async function readWmsCapabilities(baseUrl) {
+    const url = new URL(baseUrl)
+    url.searchParams.set('SERVICE', 'WMS')
+    url.searchParams.set('REQUEST', 'GetCapabilities')
+    const response = await axios.get(url.toString(), { timeout: 20000 })
+
+    if (response.status !== 200) {
+        log.error(`Failed to read GetCapabilities from ${url.toJSON()}`, response)
```

Comment:
```suggestion
        log.error('Failed to read GetCapabilities from', url, response)
```
You can give as many extra params as you want to the log function, they will be outputted as Object in the console (and you will be able to look into them, as usual)